src/response.rs

```diff
@@ -57,6 +57,41 @@ pub enum Authorization {
     Error(AuthorizationError),
 }
 
+impl Authorization {
```

Comment:
no tests?

Reply:
:scream: hold my beer

Reply:
Ok, I ended up refactoring a bit and adding quite a bit more helpers - in particular a few helpers to help with in-place modification of usage reports, which can become useful for caching. I think we can still add a few more tests, or rather asserts in the existing ones. Can you take a look?